src/controllers/routine/get-lectures-by-grade.ts

```diff
@@ -0,0 +1,78 @@
+import { PrismaClient } from '@prisma/client';
+
+import School from '../../models/school.model';
+import asyncHandler from '../../utils/asyncHandler';
+import throwError from '../../utils/throw-error';
+
+const { school } = new PrismaClient();
+
+/**
+ * @auth required
+ * @route {POST} /lecture/get_lectures_by_grade
+ * @body {grade: string, day?: string }
+ * @returns {fected lectures}
+ */
+
+const getLecturesByGrade = asyncHandler(async (req, res) => {
+  try {
+    const { grade, schoolId, day } = req.body as {
```

Comment:
instead of typecasting manually use already defined model interface

Reply:
@Prodip-Kumar-Paul it is not totally related to a model. this is more like API parameters